LibreSSL 2.7 supports newer OpenSSL APIs. Remove special handling for libreSSL.
--- modules/ssl/ssl_private.h.orig	2018-04-23 11:51:35.849536609 +0200
+++ modules/ssl/ssl_private.h	2018-04-23 11:51:49.578557354 +0200
@@ -123,7 +123,7 @@
 #define MODSSL_SSL_METHOD_CONST
 #endif
 
-#if defined(LIBRESSL_VERSION_NUMBER)
+#if 0
 /* Missing from LibreSSL */
 #if LIBRESSL_VERSION_NUMBER < 0x2060000f
 #define SSL_CTRL_SET_MIN_PROTO_VERSION          123
